File: example-java/app/controllers/ExampleServiceController.java
package controllers;

import play.api.mvc.EssentialAction;
import net.koofr.play.ws.WSController;

public class ExampleServiceController extends WSController {

  private static final ExampleServiceController instance = new ExampleServiceController();

  private ExampleServiceController() {
    super();
  }

  public Object getService() {
    return new ExampleService();
  }

  public String getScheme() {
    return "http";
  }

  public String getHost() {
    return "localhost";
  }

  public int getPort() {
    return 9000;
  }

  public String getContextPath() {
    return "example-service/";
  }

  public static EssentialAction handleGet(String path) {
    return instance.get(path);
  }

  public static EssentialAction handlePost(String path) {
    return instance.post(path);
  }

}

File: example-scala/app/controllers/ExampleServiceController.scala
package controllers

import net.koofr.play.ws._

object ExampleServiceController extends WSController {

  def getService() = {
    new ExampleService()
  }

  def getScheme() = {
    "http"
  }

  def getHost() = {
    "localhost"
  }

  def getPort() = {
    9000
  }

  def getContextPath() = {
    "example-service/"
  }

}
